### PR TITLE
Adds back bser.pdu_len.

### PR DESCRIPTION
**Pull request by dhruvsinghal**

### CODE REVIEW

**Comment by facebook-github-bot-0** [comment]
By analyzing the blame information on this pull request, we identified @sid0, @wez and @Qwait to be potential reviewers.

File: python/pywatchman/bser.c [hunk 1]
```diff
@@ -939,14 +939,11 @@ static PyObject *bser_loads_recursive(const char **ptr, const char *end,
...
-// then call bser.pdu_info on the packet.  It returns the total
-// length of the entire response that the peer is sending,
-// including the bytes already received.  This allows the client
-// to compute the data size it needs to read before it can
-// decode the data
-static PyObject *bser_pdu_info(PyObject *self, PyObject *args)
-{
+// This function parses the PDU header and provides info about the packet
+// Returns 0 if unsuccessful
+static int pdu_info_helper(PyObject *self, PyObject *args,
```
Comment (sid0) [inline python/pywatchman/bser.c]: Use `bool` in that case.

File: python/pywatchman/__init__.py [hunk 2]
```diff
@@ -553,7 +553,7 @@ def receive(self):
         if not buf[0]:
             raise WatchmanError('empty watchman response')
 
-        elen, recv_bser_version, recv_bser_capabilities = bser.pdu_info(buf[0])
+        recv_bser_version, recv_bser_capabilities, elen = bser.pdu_info(buf[0])
```
Comment (sid0) [inline python/pywatchman/__init__.py]: Any reason you juggled these around?

Comment (dhruvsinghal) [inline python/pywatchman/__init__.py]: This order seems more reasonable to me (this is the order they are present in the header).

Comment (sid0) [inline python/pywatchman/__init__.py]: OK -- might be worth splitting that into a separate commit since it is unrelated

Comment (dhruvsinghal) [inline python/pywatchman/__init__.py]: Done.

File: python/pywatchman/bser.c (hunk 1 above)
Comment (dhruvsinghal) [inline python/pywatchman/bser.c]: Done.

**Comment by wez** [comment]
Failing in a couple of ways:

```
FAIL TestBSERDump.test_pdu_len [pywatchman.pybser] (0.000s)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 327, in run
    testMethod()
  File "/home/travis/build/facebook/watchman/python/tests/tests.py", line 228, in test_pdu_len
    self.assertEqual(len(enc), self.bser_mod.pdu_len(enc))
  File "/usr/lib/python2.7/unittest/case.py", line 511, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python2.7/unittest/case.py", line 504, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: 9 != 1
```

```
pywatchman/bser.c(946) : error C2061: syntax error : identifier 'pdu_info_helper'
pywatchman/bser.c(946) : error C2059: syntax error : ';'
pywatchman/bser.c(946) : error C2059: syntax error : 'type'
pywatchman/bser.c(1005) : warning C4013: 'pdu_info_helper' undefined; assuming extern returning int
```

**Comment by facebook-github-bot-8** [comment]
@dhruvsinghal updated the pull request.

**Comment by sid0** [comment]
The second issue looks like `bool` isn't defined. Did you manage to figure this one out with @wez?

**Comment by dhruvsinghal** [comment]
Yeah, the `bool` issue is what I was concerned about. For some reason, the use of `bool` has been avoided in this file, e.g. https://github.com/facebook/watchman/blob/master/python/pywatchman/bser.c#L263 and https://github.com/facebook/watchman/blob/master/python/pywatchman/bser.c#L385

**Comment by sid0** [comment]
OK. Given that precedent if you can't figure it out quickly just switch to `int`. (If you do then you get bonus points :) )

**Comment by facebook-github-bot-7** [comment]
@dhruvsinghal updated the pull request.

**Comment by sid0** [comment]
Applied and pushed, thanks!
